pkg/kubectl/describe.go

```diff
@@ -1231,6 +1233,44 @@ func describeSecret(secret *api.Secret) (string, error) {
 	})
 }
 
+type TemplateDescriber struct {
+	clientset.Interface
+}
+
+func (td *TemplateDescriber) Describe(namespace, name string, describerSettings DescriberSettings) (string, error) {
+	temp, err := td.Extensions().Templates(namespace).Get(name)
+	if err != nil {
+		return "", err
+	}
+	return tabbedString(func(out io.Writer) error {
+		fmt.Fprintf(out, "Name:\t%v\n", temp.Name)
+		fmt.Fprintf(out, "Namespace:\t%v\n", temp.Namespace)
+		fmt.Fprintf(out, "Parameters (Count: %v):\n    Name\tDisplayName\tRequired\tType\tValue\tDescription\n", len(temp.Spec.Parameters))
+		fmt.Fprint(out, "    ----\t--------\t--------\t----\t------\t-----------\n")
+		for _, p := range temp.Spec.Parameters {
+			fmt.Fprintf(out, "    %s\t%s\t%t\t%s\t%s\t%s\n",
+				p.Name, p.DisplayName, p.Required, p.Type, p.Value, p.Description)
+		}
+		fmt.Fprintf(out, "Items (Count: %v):\n    Name\tApiGroupVersion\tKind\n", len(temp.Spec.Items))
+		fmt.Fprint(out, "    ----\t---------------\t----\n")
+		for _, i := range temp.Spec.Items {
+			decodedObj, err := runtime.Decode(runtime.UnstructuredJSONScheme, i.Raw)
+			unstructured := decodedObj.(*runtime.Unstructured)
+			if err != nil {
+				return err
+			}
+			fmt.Fprintf(out, "    %s\t%s\t%s\n", unstructured.GetName(), unstructured.GetAPIVersion(), unstructured.GroupVersionKind().Kind)
+		}
+
+		fmt.Fprintf(out, "Labels (Count: %v):\n    Key\tValue\n", len(temp.Spec.ObjectLabels))
+		fmt.Fprint(out, "    ---\t-----\n")
+		for k, v := range temp.Spec.ObjectLabels {
+			fmt.Fprintf(out, "    %s\t%s\n", k, v)
```

Comment:
You can use `labels.FormatLabels(temp.Spec.ObjectLabels)` to print this as a single line, or use `printLabelsMultilineWithIndent` to print it as multi-lines (with "    " indent).